packages/components/textarea/src/textarea.spec.ts

```diff
@@ -0,0 +1,351 @@
+import { ResizeType, Textarea, WrapType } from './textarea';
+
+describe('sl-textarea', () => {
+  let el: Textarea;
+
+  describe('defaults', () => {
+    beforeEach(async () => {
+      el = await fixture(html`<sl-textarea></sl-textarea>`);
+    });
+
+    it('should render correctly', () => {
+      expect(el).shadowDom.to.equalSnapshot();
+    });
+
+    it('should have a textarea slot', () => {
+      const slot = el.renderRoot.querySelector('slot[name="textarea"]');
+
+      expect(slot).to.not.be.null;
+      expect(slot).to.have.attribute('name', 'textarea');
+    });
+
+    it('should not be disabled', () => {
+      expect(el).not.to.have.attribute('disabled');
+    });
+
+    it('should be disabled if set', async () => {
+      el.disabled = true;
+      await el.updateComplete;
+
+      expect(el).to.have.attribute('disabled');
+    });
+
+    it('should not have a value by default', () => {
+      expect(el).not.to.have.attribute('value');
+    });
+
+    it('should have a value when set', async() => {
+      el.value = "my value";
+      await el.updateComplete;
+
+      const textarea = el.querySelector('textarea');
+
+      expect(textarea?.value).to.equal('my value');
+    });
+
+    it('should have a medium size by default', () => {
+      expect(el).to.have.attribute('size', 'md');
+    });
+
+    it('should have a large size when set', () => {
+      el.setAttribute('size', 'lg');
+
+      expect(el).to.have.attribute('size', 'lg');
+    });
+
+    it('should have a wrap property of soft by default', () => {
+      expect(el).to.have.attribute('wrap', 'soft');
+    });
+
+    it('should have a wrap property of hard when set', () => {
+      el.setAttribute('wrap', 'hard');
+
+      expect(el).to.have.attribute('wrap', 'hard');
+    });
+
+    it('should not be readonly by default', () => {
+      expect(el).not.to.have.attribute('readonly');
+    });
+
+    it('should be readonly when set', async () => {
+      el.readonly = true;
+      await el.updateComplete;
+
+      expect(el).to.have.attribute('readonly');
+    });
+
+    it('should not have an autocomplete by default', () => {
+      expect(el).not.to.have.attribute('autocomplete');
+    });
+
+    it('should not have an autocomplete when set', async() => {
+      el.autocomplete = 'off';
+      await el.updateComplete;
+
+      const textarea = el.querySelector('textarea');
+
+      expect(textarea).to.have.attribute('autocomplete', 'off');
+    });
+
+    it('should not have a placeholder by default', () => {
+      expect(el).not.to.have.attribute('placeholder');
+    });
+
+    it('should have a placeholder when set', async () => {
+      el.placeholder = 'my placeholder';
+      await el.updateComplete;
+
+      const textarea = el.querySelector('textarea');
+
+      expect(textarea).to.have.attribute('placeholder', 'my placeholder');
+    });
+
+    it('should not have a minLength by default', () => {
+      expect(el).not.to.have.attribute('minLength');
```

Comment:
does this check if there's an actual value that is being used in the component or just if the attribute is set?

Reply:
If the attribute is not set by default.

Reply:
The attribute name should be probably be `min-length` instead of `minLength`?

Reply:
We don't want to have it consistent with the native textarea? So actually it should be `minlength` instead of `minLength ` :)